ledger-api/grpc-definitions/com/daml/ledger/api/v1/completion.proto

```diff
@@ -32,6 +36,51 @@ message Completion {
   // Optional
   string transaction_id = 3;
 
+  // The application ID that was used for the submission, as described in ``commands.proto``.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Optional for historic completions where this data is not available.
+  string application_id = 6;
+
+  // The set of parties on whose behalf the commands were executed.
+  // Contains the union of ``party`` and ``act_as`` from ``commands.proto``.
+  // The order of the parties need not be the same as in the submission.
+  // Each element must be a valid PartyIdString (as described in ``value.proto``).
+  // Optional for historic completions where this data is not available.
+  repeated string act_as = 7;
+
+  // The submission ID this completion refers to, as described in ``commands.proto``.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Optional for historic completions where this data is not available.
+  string submission_id = 4;
+
+  // The submission rank of the submission this completion refers to, as described in ``commands.proto``.
+  // Must be a valid LedgerString (as described in ``value.proto``).
+  // Optional for historic completions where this data is not available.
+  string submission_rank = 5;
+
+  // The actual deduplication window used for the submission, which is derived from ``Commands.deduplication``.
+  // The ledger may convert the deduplication period into other descriptions and extend the period
+  // in implementation-specified ways.
+  // Used to audit the deduplication guarantee described in ``commands.proto``.
+  //
+  // Optional; the deduplication guarantee applies even if the completion omits the deduplication period.
+  oneof deduplication {
+    // Specifies the start of the deduplication period by a completion stream offset.
+    //
+    // Must be a valid LedgerString (as described in ``value.proto``).
+    string deduplication_offset = 8;
```

Comment:
@andreaslochbihler-da : is the gap in field numbers intentional or due to a copy-past case?

Also: did you consider defining a DeduplicationPeriod message and reusing that here and in the `SubmitterInfo`?

Reply:
Field numbers 6 and 7 are already assigned further up (`applicationId` and `actAs`). But given that we're introducing them as part of this PR anyway, it makes sense to assign them consecutively. Changed.

Reply:
I didn't introduce a separate `DeduplicationPeriod` message for symmetry with `Commands`, where we decided that such a message would break backwards compatibility.

Reply:
Thanks for reminding me of the backwards-compatibility issue. I forgot about it. Makes sense.